synapse/util/events.py

```diff
@@ -27,3 +30,31 @@ def generate_fake_event_id() -> str:
         A string intended to look like an event ID, but with no actual meaning.
     """
     return "$" + random_string(43)
+
+
+def get_plain_text_topic_from_event_content(content: JsonDict) -> Optional[str]:
+    """
+    Given the content of an m.room.topic event returns the plain text topic
+    representation if any exists.
+
+    Returns:
+        A string representing the plain text topic.
+    """
+    topic = content.get("topic")
+
+    m_topic = content.get("m.topic")
+    if not m_topic:
+        return topic
+
+    m_text = m_topic.get("m.text")
+    if not m_text:
+        return topic
+
+    representation = next(
+        (r for r in m_text if "mimetype" not in r or r["mimetype"] == "text/plain"),
+        None,
+    )
+    if not representation or "body" not in representation:
+        return topic
```

Comment:
```suggestion
    # Find the first `text/plain` topic ("Receivers SHOULD use the first representation in the array that they understand.")
    representation = next(
        (r for r in m_text if "mimetype" not in r or r["mimetype"] == "text/plain"),
        None,
    )
    if not representation or "body" not in representation:
        return topic
```

If we really wanted to be more robust, we should try to skip to the next `text/plain` entry that may have a `body`.

The [spec](https://spec.matrix.org/v1.15/client-server-api/#mroomtopic) says: "Receivers SHOULD use the first representation in the array that they understand."

(includes scrutinizing check for non-string `body`)
```suggestion
    # Find the first `text/plain` topic ("Receivers SHOULD use the first representation in the array that they understand.")
    representation = next(
        (
            r
            for r in m_text
            if ("mimetype" not in r or r["mimetype"] == "text/plain")
            and "body" in r
            # Scrutinize user input
            and isinstance(r["body"], str)
        ),
        None,
    )
    if not representation:
        return topic
```

Reply:
Have switched to the second suggestion but committed it manually because I had already extracted the constants before.